### PR TITLE
Fix uri on pc

### PR DESCRIPTION
**Pull request by javerbukh**
### Description

This pull request is to address a problem when loading data from a windows computer. The `os.path.sep` is different on a windows and linux computer, which does not work for our example notebooks.

Fixes #<Issue Number>

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by kecnry — APPROVED** [review]

File: jdaviz/utils.py [hunk 1]
```diff
@@ -493,8 +493,7 @@ def download_uri_to_path(possible_uri, cache=None, local_path=os.curdir, timeout
 
         if local_path is None:
             # if not specified, this is the default location:
-            local_path = os.path.join(os.getcwd(), parsed_uri.path.split(os.path.sep)[-1])
-
+            local_path = os.path.join(os.getcwd(), parsed_uri.path.split('/')[-1])
```
Comment (pllim) [inline jdaviz/utils.py:496]: Why was this not caught in CI? Is this not tested on Windows?

**Comment by bmorris3** [comment]
@javerbukh Just checking that this line should remain the same, since in this case the `parsed_uri.path` is actually a path?

https://github.com/spacetelescope/jdaviz/blob/bc5dcbc801d04c048b93d9895fa881720da128e1/jdaviz/utils.py#L480

File: jdaviz/utils.py (hunk 1 above)
Comment (bmorris3) [reply jdaviz/utils.py:496]: I _might_ see what went wrong now, thanks @javerbukh. @pllim – it looks like urllib's `parse` method is for URLs and always uses `/`. This is a block in the code where the `parsed_uri.path` is parsed as a URL, which shouldn't be OS specific. Does that make sense?

Comment (pllim) [reply jdaviz/utils.py:496]: So, sounds like a new test needs to be added so this case is covered. And this new test should fail on `main` but pass with this patch. Correct?

Comment (javerbukh) [reply jdaviz/utils.py:496]: Do we have a test already of loading data with a windows type path @pllim ? If not, do you have an example of setting up a test like that?

**Comment by javerbukh** [comment]
@bmorris3 I think you are correct that that should stay the same.

File: jdaviz/utils.py (hunk 1 above)
Comment (pllim) [reply jdaviz/utils.py:496]: Given that most of the devs work off OSX, probably not. Here's an idea:

1. Find a small test that already use a data file (preferable not remote).
2. When you detect the test is running on Windows, force forward slash, maybe like this: `str(p.absolute()).replace("/", "\\")` (pathlib has `as_posix()` but it does not use forward slash when I use git bash shell)
3. See if this fails on `main`. You want it to fail on `main`.
4. See if your patch fixes it.

**Comment by javerbukh** [comment]
@pllim You were correct that the CI should be throwing errors. I ran `pytest -s jdaviz/tests/test_utils.py --remote-data` on my windows machine and it did in fact error on `main`. With these changes, that should be fixed. I can create a follow up ticket for finding a way to check remote data tests are run on multiple operating systems. 

@bmorris3 , sorry, my first response to you was incorrect. That line does need to be changed.

**Comment by pllim** [comment]
We could easily enable remote data on Windows job. But that does mean 2x server hit per run.

**Comment by kecnry** [comment]
I'm not too worried about regression coverage since this would be very hard to sneak back in accidentally.  But if we think there is more windows-specific remote data handling that should get coverage, then that may be worth it? 🤷‍♂️ 

Looks like URI support (#2875) was milestoned to 3.11, so maybe this should do the same and just be appended to that same changelog entry.

File: jdaviz/utils.py (hunk 1 above)
Comment (bmorris3) [reply jdaviz/utils.py:496]: +1 to @pllim's plan!

Comment (javerbukh) [reply jdaviz/utils.py:500]: I tried that and it does now work, feel free to test on your own windows machines with `pytest -s jdaviz/tests/test_utils.py --remote-data` .

Comment (javerbukh) [reply jdaviz/utils.py:500]: Jira ticket for adding remote data CI tests on Windows OS https://jira.stsci.edu/browse/JDAT-4596

Comment (pllim) [reply jdaviz/utils.py:500]: Should we just slap remote data on Windows job as part of this PR and take it back out when that other ticket is done?

Comment (pllim) [reply jdaviz/utils.py:500]: Would look something like this: https://github.com/spacetelescope/jdaviz/pull/2924

**Review by pllim — APPROVED** [review]
Windows CI is green even with remote data for this patch, so LGTM. Thanks!

File: jdaviz/utils.py (hunk 1 above)
Comment (javerbukh) [reply jdaviz/utils.py:500]: Added that to the CI and it looks like it is fixed in this PR. Thank you for the example!

**Pull request merged**